src/pages/userManagement/UserManagerCollaborationEditTable.jsx

```diff
@@ -0,0 +1,222 @@
+// import CollaborationsDialog from '../../components/cards/collaborations/CollaborationsDialog';
+
+export default function UserManagersCollaborationEditTable({
+  data,
+  loading,
+  error,
+  refresh,
+}) {
+  const {
+    patchCollaboration,
+    loading: patchLoading,
+    error: patchError,
+    setError: patchSetError,
+    success: patchSuccess,
+    setSuccess: patchSetSuccess,
+  } = usePatchCollaboration();
+  async function processRevoke(collaboration) {
+    const collaborationData = {
+      op: 'replace',
+      path: '/view_permission',
+      value: 'revoked',
+    };
+    const response = await patchCollaboration(
+      get(collaboration, 'guid'),
+      collaborationData,
+    );
+  }
+
+  function getNthAlphabeticalMemberObjAndMakeLodashReady(obj, n) {
+    // derived from https://stackoverflow.com/questions/39483677/how-to-get-first-n-elements-of-an-object-using-lodash
+    const memberObject = Object.keys(obj)
+      .sort()
+      .slice(n - 1, n)
+      .reduce((memo, current) => obj[current], {});
+
+    // Convert empty strings to undefined so that they get handled appropriately by lodash get later omg this is so silly
+    if (get(memberObject, 'full_name') === '')
+      memberObject.full_name = undefined;
+
+    return memberObject;
+  }
+  function transformData(inputData) {
+    if (!inputData || inputData.length === 0) return null;
+    return inputData.map(entry => ({
+      // Note: the collaboration API call returned a members OBJECT instead of array of objects, which made some tranformation gymnastics here necessary
+      guid: get(entry, 'guid'),
+      userOne: get(
+        getNthAlphabeticalMemberObjAndMakeLodashReady(
+          get(entry, 'members'),
+          1,
+        ),
+        'full_name',
+        get(
+          getNthAlphabeticalMemberObjAndMakeLodashReady(
+            get(entry, 'members'),
+            1,
+          ),
+          'email',
+        ),
+      ),
+      userTwo: get(
+        getNthAlphabeticalMemberObjAndMakeLodashReady(
+          get(entry, 'members'),
+          2,
+        ),
+        'full_name',
+        get(
+          getNthAlphabeticalMemberObjAndMakeLodashReady(
+            get(entry, 'members'),
+            2,
+          ),
+          'email',
+        ),
+      ),
+      viewStatusOne: get(
+        getNthAlphabeticalMemberObjAndMakeLodashReady(
+          get(entry, 'members'),
+          1,
+        ),
+        'viewState',
+      ),
+      viewStatusTwo: get(
+        getNthAlphabeticalMemberObjAndMakeLodashReady(
+          get(entry, 'members'),
+          2,
+        ),
+        'viewState',
+      ),
+      // editStatusOne: get(
+      //   getNthAlphabeticalMemberObjAndMakeLodashReady(get(entry, 'members'), 1),
+      //   'editState',
+      // ),
+      // editStatusTwo: get(
+      //   getNthAlphabeticalMemberObjAndMakeLodashReady(get(entry, 'members'), 2),
+      //   'editState',
+      // ),
+    }));
+  }
+  data = transformData(data);
```

Comment:
you never want to reassign a variable to itself if you can avoid it. _especially_ if that variable is a prop